src/raftstore/store/store.rs

```diff
@@ -319,6 +331,54 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         self.peer_cache.clone()
     }
 
+    fn poll_snapshot_status(&mut self) {
+        if self.sent_snapshot_count == 0 {
+            return;
+        }
+
+        // Poll all snapshot messages and handle them.
+        loop {
+            match self.snapshot_status_receiver.try_recv() {
+                Ok(SnapshotStatusMsg { region_id, to_peer_id, status }) => {
+                    // Report snapshot status to the corresponding peer.
+                    self.report_snapshot_status(region_id, to_peer_id, status);
+                }
+                Err(TryRecvError::Empty) => {
+                    // The snapshot status receiver channel is empty
+                    return;
+                }
+                Err(e) => {
+                    error!("{} unexpected error {:?} when receive from snapshot channel",
+                           self.tag,
+                           e);
+                    return;
+                }
+            }
+        }
+    }
+
+    fn report_snapshot_status(&mut self, region_id: u64, to_peer_id: u64, status: SnapshotStatus) {
+        self.sent_snapshot_count -= 1;
```

Comment:
what I mean if we meet some error, the snapshot_count is not added, but we still get the snapshot report message.